File: Templates/sliding_window_variable_longest.cpp
#include <vector>
#include <algorithm>

// Placeholder for user-defined helpers
bool invalid(const std::vector<int>& window);
int optimal(const std::vector<int>& window);

int sliding_window_flexible_longest(const std::vector<int>& input) {
    int answer = 0;
    std::vector<int> window;
    int left = 0;
    for (int right = 0; right < (int)input.size(); ++right) {
        window.push_back(input[right]);
        while (invalid(window)) {
            int left_elem = input[left];
            auto it = std::find(window.begin(), window.end(), left_elem);
            if (it != window.end()) {
                window.erase(it);
            }
            ++left;
        }
        answer = std::max(answer, optimal(window));
    }
    return answer;
}

File: Templates/sliding_window_variable_longest.py
def sliding_window_flexible_longest(input):
    anwser = 0
    window = []

    left = 0

    for right in range(len(input)):
        window.append(input[right])
        while invalid(window):        # update left until window is valid again
            window.remove(input[left])
            left += 1
        anwser = max(anwser, optimal(window))        # window is guaranteed to be valid here

    return anwser
